src/main/java/com/github/javafaker/CPF.java

```diff
@@ -0,0 +1,65 @@
+package com.github.javafaker;
+
+import com.github.javafaker.shared.CPFUtils;
+
+import java.util.Random;
+
+/**
+ * Partial code copy from https://github.com/jrjuniorsp/GeradorValidadorCPFCNPJ/blob/master/src/com/jrmobile/service/GeradorCPF.java
+ */
+public class CPF {
+
+    private Faker faker;
+
+    protected CPF(Faker faker) {
+        this.faker = faker;
+    }
+
+    /**
+     * Return valid and formatted
+     * @see com.github.javafaker.shared.CPFUtils#unformatCPF(String) to unformat
+     * @return a valid CPF
+     */
+    public String valid() {
+        return generateCPF(true);
+    }
+
+    /**
+     * Return invalid and formatted
+     * @see com.github.javafaker.shared.CPFUtils#unformatCPF(String) to unformat
+     * @return an invalid CPF
+     */
+    public String invalid() {
+        return generateCPF(false);
+    }
+
+    public String generateCPF(boolean valid) {
+        String partial = "";
+        String cpf = "";
+        Integer number;
+
+        for (int i = 0; i < 9; i++) {
+            number = new Integer((int) (Math.random() * 10));
+            partial += number.toString();
+        }
+
+        if(valid) {
+            cpf = partial + CPFUtils.calculateVerificationDigit(partial);
+        } else {
+            String twoFakeDigits = String.valueOf(new Random().nextInt(9 + 1)) + String.valueOf(new Random().nextInt(9 + 1));
```

Comment:
Isn't there a chance that this could also return a valid checkSum? 1 in 100 chance possibly?

Reply:
You mean like this: https://github.com/hibernate/hibernate-validator/blob/master/engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/ModCheckValidator.java